staging/src/k8s.io/api/batch/v1/generated.proto

```diff
@@ -29,6 +29,89 @@ import "k8s.io/apimachinery/pkg/runtime/schema/generated.proto";
 // Package-wide variables from generator "generated".
 option go_package = "v1";
 
+// CronJob represents the configuration of a single cron job.
+message CronJob {
+  // Standard object's metadata.
+  // More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+  // +optional
+  optional k8s.io.apimachinery.pkg.apis.meta.v1.ObjectMeta metadata = 1;
+
+  // Specification of the desired behavior of a cron job, including the schedule.
+  // More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+  // +optional
+  optional CronJobSpec spec = 2;
+
+  // Current status of a cron job.
+  // More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+  // +optional
+  optional CronJobStatus status = 3;
+}
+
+// CronJobList is a collection of cron jobs.
+message CronJobList {
+  // Standard list metadata.
+  // More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+  // +optional
+  optional k8s.io.apimachinery.pkg.apis.meta.v1.ListMeta metadata = 1;
+
+  // items is the list of CronJobs.
+  repeated CronJob items = 2;
+}
+
+// CronJobSpec describes how the job execution will look like and when it will actually run.
+message CronJobSpec {
+  // The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
+  optional string schedule = 1;
```

Comment:
Isn't this required?

Reply:
You're looking at generated proto which wasn't written manually. If you check the pre-existing proto you'll notice it looks identical https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/batch/v1beta1/generated.proto#L65. But if you check `staging/src/k8s.io/api/batch/v1/types.go` in this PR (https://github.com/kubernetes/kubernetes/pull/98965/files#diff-238c6caec0e3aac5feb3c52bd6a356ea4d579a7f97cf50d741abee5fd95d0c6bR269) you'll notice it's not optional. Additionally, this PR as described in the KEP is not changing any parts of the batch/v1beta1 API, but promoting this API as is with a small addition of `LastSuccessfulTime`.

Reply:
it is not required to decode the protobuf object; it is required in API validation. if we mark it required here, then proto decoding would fail and we wouldn't have the chance to reach API validation and provide more useful messages.

Reply:
That is good to know. Sorry for the noise.